### PR TITLE
Add add on association

### PR DESCRIPTION
**Pull request by project-eutopia**
There is a has_many :through relation from `LineItem` to `AddOn`, but the reverse isn't present.  This fixes that.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -4,13 +4,11 @@ source 'http://rubygems.org'
 gem "spree_auth_devise", github: "spree/spree_auth_devise", branch: '1-3-stable'
 
 # we use line_item_description_text which is in newer sprees
-gem 'spree', github: 'freerunningtech/spree', branch: '1-3-stableish'
+gem 'spree', github: 'degica/spree', branch: '1-3-rebased-clean'
 
 group :development, :test do
-  gem 'fuubar'
```
Comment (shioyama) [inline Gemfile]: You're removing `fuubar`, is that related to this PR? If not please make it a separate commit.

Comment (project-eutopia) [inline Gemfile]: @shioyama Alright, I made the first 3 commits that fix dependencies only into a separate PR #2.

Comment (shioyama) [inline Gemfile]: Thanks, merged #2, can you remove these changes here? Then I can merge this.

**Comment by project-eutopia** [comment]
@shioyama Thanks for the merge on #2.  This PR is now just a couple short changes.

**Comment by degicat** [comment]
@k2nr Please review this.

**Comment by k2nr** [comment]
LGTM :+1:

**Comment by project-eutopia** [comment]
@k2nr Thanks!
@shioyama Could you merge this for me please?

**Comment by shioyama** [comment]
@project-eutopia done, thanks!
